internal/security/proxy/requestor.go

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.1.0
+ *******************************************************************************/
+package proxy
+
+import (
+	"net/http"
+)
+
+type Requestor interface {
```

Comment:
This interface represents targets that the requests send to. In the unit testing these functions can be implemented in a way that a remote server is not needed.

Reply:
I still don't think this is an appropriate usage of interfaces in Go, because all of the methods are just returning elements from the struct. To mock this in the unit testing, just create a different `*http.Client` in the struct that is using the mocked http client. This should be discussed in the WG meeting

Reply:
Yes we create different *http.Client in the testing structure and use it for unit testing.